src/layers.c

```diff
@@ -39,34 +55,13 @@ layers_arrange(struct output *output)
 
 	int nr_layers = sizeof(output->layers) / sizeof(output->layers[0]);
 	for (int i = 0; i < nr_layers; i++) {
-		struct lab_layer_surface *lab_layer_surface;
-
-		/*
-		 * First we go over the list of surfaces that have
-		 * exclusive_zone set (e.g. statusbars) because we have to
-		 * determine the usable area before processing regular layouts.
-		 */
-		wl_list_for_each(lab_layer_surface, &output->layers[i], link) {
-			struct wlr_scene_layer_surface_v1 *scene_layer_surface =
-				lab_layer_surface->scene_layer_surface;
-			if (scene_layer_surface->layer_surface->current.exclusive_zone) {
-				wlr_scene_layer_surface_v1_configure(
-					scene_layer_surface, &full_area, &usable_area);
-			}
-		}
-
-		/* Now we process regular layouts */
-		wl_list_for_each(lab_layer_surface, &output->layers[i], link) {
-			struct wlr_scene_layer_surface_v1 *scene_layer_surface =
-				lab_layer_surface->scene_layer_surface;
-			if (!scene_layer_surface->layer_surface->current.exclusive_zone) {
-				wlr_scene_layer_surface_v1_configure(
-					scene_layer_surface, &full_area, &usable_area);
-			}
-		}
+		struct wlr_scene_tree *layer = output->layer_tree[i];
+		arrange_one_layer(output, &full_area, &usable_area, layer);
 
-		wlr_scene_node_set_position(&output->layer_tree[i]->node,
-			scene_output->x, scene_output->y);
+		/* TODO: Is this really needed? */
```

Comment:
Guess the `TODO` can be removed now, it is needed in case the output position in the server layout changes.
`wlr_scene_node_set_position()` has an internal check which compares the old values vs the new ones and early returns in case they are the same so this is pretty cheap.

The alternative would be to listen to layout changes here as well and then act accordingly.

Reply:
Done